packages/sync/src/config/AuthContextProvider.ts

```diff
@@ -0,0 +1,10 @@
+
+export interface AuthContext {
+  header: any;
+  token: string;
+}
+
+/**
+ * Interface for AuthContextProvider
+ */
+export type AuthContextProvider = () => Promise<AuthContext>;
```

Comment:
Not 100% sure but I think we can avoid having this type and return `Promise<AuthContext>` explicitly above.

Reply:
Ah.. checked code locally. I was wrong